lib/ts/fetch.ts

```diff
@@ -409,11 +424,9 @@ export async function onUnauthorisedResponse(
                     return { result: "RETRY" };
                 }
 
-                await saveRefreshAttempt();
-
```

Comment:
don't we need to update last access token update time after a successful refresh? If the refresh succeeds, then it will call the front-token update which will do this, but if the refresh fails (for example if the user is logged out), then it won't save this last access token update time and then it will keep calling refresh on each page load?

Reply:
- if refresh returns anything that updates the front token (200, or 401 clearing tokens), FrontToken.setItem will save the update
- if refresh returns 401 that didn't update tokens (i.e.: FE thinks it has tokens, but BE didn't get them), then we clear FrontToken on line 473
- if we errored out for other reasons (i.e., we got a 500 response, and it didn't update any tokens), then I don't think calling the refresh again on the next page load is a problem

Reply:
One of the developer experience things is that the frontend UI should show even if the backend integration is not done. In this case, when the frontend loads up for the first time, it will get a 404 from the refresh endpoint, and each time the developer reloads the page or doesSessionExist is called, it will call the refresh API (until they have finished the backend integration). This might result in several calls to the refresh API in one go which can be weird.

If I am right, and if there is no harm in moving this to after the refresh has been called (even if it returned a 500 error), then please move it after.

Reply:
@porcellus ^^

Reply:
I think that's only appropriate in the 404 case
If:
- the FE cookies have been cleared (e.g.: in safari)
- the BE returns a 500 and we save the update cookie
Then on the next reload we will not call refresh and consider the user logged out even though they have valid tokens in the BE set cookies.

This is an edge-case, but can occur in production instead of a minor weird behaviour during setup.

Reply:
We need to know how many times refresh will get called if the backend is not started. We should also write a test for this case to make sure that refresh is not called 10 times or something in this case. Maybe 1-3 times is fine;

Reply:
(Keeping this comment open) - but the problem is resolved.

Reply:
in the test app we call 1-2 times depending on what page was opened